pkg/controllers/v1/mpi_job_controller.go

```diff
@@ -776,6 +817,30 @@ func (c *MPIJobController) getOrCreateWorker(mpiJob *kubeflow.MPIJob) ([]*corev1
 		return workerPods, nil
 	}
 
+	// Remove Pods when replicas are scaled down
+	selector, err := workerSelector(mpiJob.Name)
+	if err != nil {
+		return nil, err
+	}
+	podFullList, err := c.podLister.List(selector)
+	if err != nil {
+		return nil, err
+	}
+	if len(podFullList) > int(*workerReplicas) {
+		for _, pod := range podFullList {
+			indexStr := strings.TrimLeft(pod.Name, fmt.Sprintf("%s-", workerPrefix))
+			index, err := strconv.Atoi(indexStr)
+			if err == nil {
+				if index >= int(*workerReplicas) {
+					err = c.kubeClient.CoreV1().Pods(pod.Namespace).Delete(pod.Name, &metav1.DeleteOptions{})
```

Comment:
Do we need to expose delete options?

Reply:
Do you mean `&metav1.DeleteOptions{}`?

Reply:
Yes

Reply:
I would prefer keep the default delete options before we reached a consensus on how to expose these delete options. If we expose this by adding fields on mpijob crd, that could lead to a bigger modification to the code.

Reply:
I think so.

Reply:
Sounds good. We can add when needed later.